pkg/api/rest/delete.go

```diff
@@ -91,3 +91,16 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx api.Context, obj runtime.Obje
 	objectMeta.DeletionGracePeriodSeconds = options.GracePeriodSeconds
 	return true, false, nil
 }
+
+// MarkImmediateDeletion set the obj's DeletionGracePeriodSeconds to 0, and set the DeletionTimestamp to now.
+func MarkImmediateDeletion(strategy RESTDeleteStrategy, obj runtime.Object) (err error) {
```

Comment:
Name should say something about finalizers, this is not really immediate.

Why did you put it in this package? I would put it in the file it's used, and make it private.